### PR TITLE
gossip: wallclock update in during_housekeeping

### PR DESCRIPTION
**Pull request by ravyu-jump**
Since we don't need nanosecond precision for gossip pushes, we can reduce frequency of calls to `fd_log_wallclock()` in gossip tile by moving it to the `during_housekeeping()` callback.

### CODE REVIEW

File: src/app/fdctl/run/tiles/fd_gossip.c [hunk 1]
```diff
@@ -356,6 +356,15 @@ gossip_signer( void *        signer_ctx,
   fd_keyguard_client_sign( ctx->keyguard_client, signature, buffer, len, sign_type );
 }
 
+static void
+during_housekeeping( void * _ctx ) {
```
Comment (mmcgee-jump) [inline src/app/fdctl/run/tiles/fd_gossip.c:360]: This can just be `fd_gossip_tile_ctx_t * ctx`

File: src/app/fdctl/run/tiles/fd_gossip.c [hunk 2]
```diff
@@ -356,6 +356,15 @@ gossip_signer( void *        signer_ctx,
...
+  fd_gossip_tile_ctx_t * ctx = (fd_gossip_tile_ctx_t *)_ctx;
+
+  /* TODO: check if this can be done more infrequently */
```
Comment (mmcgee-jump) [inline src/app/fdctl/run/tiles/fd_gossip.c:363]: Think you can remove the comment?

File: src/app/fdctl/run/tiles/fd_gossip.c [hunk 3]
```diff
@@ -812,6 +820,7 @@ populate_allowed_fds( fd_topo_t const *      topo,
 #define STEM_CALLBACK_CONTEXT_ALIGN alignof(fd_gossip_tile_ctx_t)
 
 #define STEM_CALLBACK_AFTER_CREDIT after_credit
+#define STEM_CALLBACK_DURING_HOUSEKEEPING during_housekeeping
```
Comment (mmcgee-jump) [inline src/app/fdctl/run/tiles/fd_gossip.c:823]: nit: Can you keep these aligned :)

File: src/app/fdctl/run/tiles/fd_gossip.c (hunk 1 above)
Comment (ravyu-jump) [reply src/app/fdctl/run/tiles/fd_gossip.c:360]: oh nice didn't catch that

File: src/app/fdctl/run/tiles/fd_gossip.c [hunk 4]
```diff
@@ -811,10 +815,11 @@ populate_allowed_fds( fd_topo_t const *      topo,
 #define STEM_CALLBACK_CONTEXT_TYPE  fd_gossip_tile_ctx_t
 #define STEM_CALLBACK_CONTEXT_ALIGN alignof(fd_gossip_tile_ctx_t)
 
-#define STEM_CALLBACK_AFTER_CREDIT after_credit
-#define STEM_CALLBACK_BEFORE_FRAG  before_frag
-#define STEM_CALLBACK_DURING_FRAG  during_frag
-#define STEM_CALLBACK_AFTER_FRAG   after_frag
+#define STEM_CALLBACK_AFTER_CREDIT          after_credit
+#define STEM_CALLBACK_DURING_HOUSEKEEPING   during_housekeeping
```
Comment (mmcgee-jump) [inline src/app/fdctl/run/tiles/fd_gossip.c:819]: extra extra nit: still not quite aligned. You have 3 whitespaces here but just need one.

**Review by mmcgee-jump — APPROVED** [review]

File: src/app/fdctl/run/tiles/fd_gossip.c (hunk 4 above)
Comment (ravyu-jump) [reply src/app/fdctl/run/tiles/fd_gossip.c:819]: I didn't like how cramped this housekeeping line looked lol. I'll fix that

**Review by mmcgee-jump — DISMISSED** [review]

**Comment by ravyu-jump** [comment]
@mmcgee-jump messed up the force push the whitespace changes should be included now

**Review by mmcgee-jump — APPROVED** [review]

**Pull request merged**